### PR TITLE
util-misc: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Task #6345

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Update `UtilMiscParseSizeStringTest01` in util-misc unit tests

### CODE REVIEW

File: src/util-misc.c [hunk 1]
```diff
@@ -933,235 +645,136 @@ static int UtilMiscParseSizeStringTest01(void)
...
+    FAIL_IF(ParseSizeString(str, &result) > 0);
+    FAIL_IF(result != 10.5 * 1024 * 1024 * 1024);
 
     /* Should fail on unknown units. */
-    if (ParseSizeString("32eb", &result) > 0) {
-        goto error;
-    }
+    FAIL_IF(ParseSizeString("32eb", &result) > 0);
 
     return 1;
```
Comment (inashivb) [inline src/util-misc.c:776]: This should be made `PASS`

**Review by inashivb — CHANGES_REQUESTED** [review]
Hi, Hadiqa!
Nice job on your first PR. :)
I have one change requested from me inline. Thank you!

File: src/util-misc.c (hunk 1 above)
Comment (hadiqaalamdar) [reply src/util-misc.c:776]: understood, I'll make the change and create a new PR. Thanks!

Comment (hadiqaalamdar) [reply src/util-misc.c:776]: @inashivb should I let the name of the new PR be the same as this one? And should I add the link of this PR in the description of the new one?

Comment (jufajardini) [reply src/util-misc.c:776]: > @inashivb should I let the name of the new PR be the same as this one?

Yes, please, create a new PR with the same title as this one, adding a version to it (so: `util-misc: Convert unittests to new FAIL/PASS API - v2`)

> And should I add the link of this PR in the description of the new one?

Yes, and then you can close this one with a note that work is continued on the new PR, and add a link :)

Comment (hadiqaalamdar) [reply src/util-misc.c:776]: will do, thanks!

**Comment by hadiqaalamdar** [comment]
Work is continued on https://github.com/OISF/suricata/pull/9656

**Pull request closed without merging**